### PR TITLE
:art: Improve the nullish value usage around the codebase

### LINKED ISSUE #5094: Improve the nullish value usage around the codebase
### This is not a Bug Report, Feature Request, or related to Documentation

- [X] I have searched the existing issues

### Is there an existing issue for this?

- [X] I have searched the existing issues

### What's up?

In our codebase we are using mixed of  **nullish** values, sometimes `null` and sometimes `undefined`. Both have different behavior and can result into different side-effects. I suggest to use this consistent approach throughout the codebase. 

1. Use Typescript **optional** parameter feature where ever possible. 
2. Try to avoid `null` type and if needed use `undefined` which is aligned with **optional** attribute behavior 
3. Whenever checking for **nullish** values, always check for both `null` and `undefined` to avoid null pointer reference. 
4. When it's a need of **nullish** value, use `undefined` as value instead of `null`.

### Examples/References

Talk by Tony Hoare:
https://www.infoq.com/presentations/Null-References-The-Billion-Dollar-Mistake-Tony-Hoare/

Key Takeaways: 

- Null references have historically been a bad idea
- Early compilers provided opt-out switches for run-time checks, at the expense of correctness
- Programming language designers should be responsible for the errors in programs written in that language

### PR DESCRIPTION
**Pull request by nazarhussain**
## Description

Update all the usage of `nullish` values around the codebase as per the guideline in the issue. 

Fixes #5094

## Type of change

- [x] Improvement 

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [x] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [x] I ran `npm run build` with success.
- [x] I have tested the built `dist/web3.min.js` in a browser.
- [x] I have tested my code on the live network.
- [x] I have checked the Deploy Preview and it looks correct.
- [x] I have updated the `CHANGELOG.md` file in the root folder.

### CODE REVIEW

**Comment by nazarhussain** [comment]
The changes related to doc comments blocks appears because of running `yarn lint:fix`

**Review by avkos — APPROVED** [review]

File: packages/web3-errors/src/web3_error_base.ts [hunk 1]
```diff
@@ -36,6 +36,10 @@ export abstract class Web3Error extends Error {
 	public static convertToString(value: unknown, unquotValue = false) {
 		if (value === undefined) return 'undefined';
 
+		// Using "null" value intentionally for validation
+		// eslint-disable-next-line no-null/no-null
+		if (value === null) return 'undefined';
```
Comment (jdevcs) [inline packages/web3-errors/src/web3_error_base.ts]:
```suggestion
		// Using "null" value intentionally for validation
		// eslint-disable-next-line no-null/no-null
		if (value === null || value === undefined ) return 'undefined';
```

File: packages/web3-eth/src/types.ts [hunk 2]
```diff
@@ -65,7 +65,7 @@ export interface ReceiptInfo {
 	readonly cumulativeGasUsed: Numbers;
 	readonly gasUsed: Numbers;
 	readonly effectiveGasPrice?: Numbers;
-	readonly contractAddress: Address | null;
+	readonly contractAddress: Address;
```
Comment (jdevcs) [inline packages/web3-eth/src/types.ts]:
```suggestion
	readonly contractAddress?: Address;
```

**Review by jdevcs — CHANGES_REQUESTED** [review]
LGTM, only minor changes required.

**Comment by jdevcs** [comment]
and also some tests are failing so needs to be fixed in this PR.

**Review by jdevcs — APPROVED** [review]

**Review by luu-alex — APPROVED** [review]

**Pull request merged**